### ISSUE TITLE
Error message persists for package imports

### ISSUE THREAD

**Issue by minsa110** [microsoft/pylance-release#2246]
## Environment data

-   VS Code version: 1.63.2
-   Jupyter Extension version (available under the Extensions sidebar): v2021.11.1001550889
-   Python Extension version (available under the Extensions sidebar): v2021.12.1559732655
-   OS (Windows | Mac | Linux distro) and version: Windows
-   Python and/or Anaconda version: Python 3.8.12 / Conda 4.11.0
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): conda
-   Jupyter server running: Local

## Expected behaviour

Error message about import failure should be gone after I've installed it on my conda environment since package is now importable.

## Actual behaviour

https://user-images.githubusercontent.com/10833267/149375676-4f70686c-3261-4337-b6cc-b36daad79ede.mp4

## Steps to reproduce:

1. Run the cell with import statements
2. Imports fail since they're not installed in my conda environment (as expected)
3. Install the package on my conda environment
4. Run the cell again
5. Imports work (as expected), but the error message persists (NOT expected)
6. Reopening the notebook does not fix it
7. Reopening VS Code fixes it

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
Jupyter Extension Version: 2021.11.1001550889.
Python Extension Verison: 2021.12.1559732655.
Info 08:47:37: Experimentation service retrieved: [object Object]
User belongs to experiment group 'jupyterTestcf'
User belongs to experiment group 'jupyterEnhancedDataViewer'
Info 08:47:37: ZMQ install verified.
Info 08:47:40: Experiment status for python is {"enabled":true,"optInto":[],"optOutFrom":[]}
Info 08:47:40: Setting setActiveController for /mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science/Titanic.ipynb
Info 08:47:41: KernelProvider switched kernel to id = .jvsc74a57bd02c13d1dba576189d0c1cbb18bd79e08368b732b610f52891218258264bcb2590./home/minsa110/anaconda3/envs/py38/python./home/minsa110/anaconda3/envs/py38/python.-m#ipykernel_launcher
Info 08:47:41: Starting Notebook in kernel.ts id = .jvsc74a57bd02c13d1dba576189d0c1cbb18bd79e08368b732b610f52891218258264bcb2590./home/minsa110/anaconda3/envs/py38/python./home/minsa110/anaconda3/envs/py38/python.-m#ipykernel_launcher
Info 08:47:41: Creating raw notebook for /mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science/Titanic.ipynb
Info 08:47:41: Getting preferred kernel for /mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science/Titanic.ipynb
Info 08:47:41: Computing working directory /mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science/Titanic.ipynb
Info 08:47:41: Process Execution: > ~/anaconda3/envs/py38/bin/python -m pip list
> ~/anaconda3/envs/py38/bin/python -m pip list
Info 08:47:41: Starting raw kernel py38 (Python 3.8.12)
Info 08:47:41: installMissingDependencies ~/anaconda3/envs/py38/bin/python
Info 08:47:41: IPykernel found previously in this enviornment ~/anaconda3/envs/py38/bin/python
Info 08:47:41: Kernel launching with ports 9015,9016,9017,9018,9019. Start port is 9000
Info 08:47:41: Launching kernel daemon for Python 3.8.12 64-bit ('py38': conda) # ~/anaconda3/envs/py38/bin/python
Info 08:47:41: Creating daemon process for ~/anaconda3/envs/py38/bin/python with env variables count 65
Info 08:47:41: Process Execution: > ~/anaconda3/envs/py38/bin/python -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> ~/anaconda3/envs/py38/bin/python -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
Info 08:47:41: Creating controller for jupyter-notebook with interpreter ~/miniconda3/envs/golden-scenario/bin/python
Info 08:47:41: Find preferred kernel for /mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science/Titanic.ipynb with metadata {"interpreter":{"hash":"7118164a812f03e7abaca324470c207028af9f5eb97c564b01cda75823eeb50c"},"kernelspec":{"display_name":"Python 3.7.10 64-bit ('base': conda)","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.12"},"metadata":{"interpreter":{"hash":"7118164a812f03e7abaca324470c207028af9f5eb97c564b01cda75823eeb50c"}}} & preferred interpreter ~/miniconda3/envs/golden-scenario/bin/python
Info 08:47:41: findKernel found Python 3.10.0 64-bit
Info 08:47:41: PreferredConnection: .jvsc74a57bd097cc609b13305c559618ec78a438abc56230b9381f827f22d070313b9a1f3777./bin/python3.10./bin/python3.10.-m#ipykernel_launcher found for NotebookDocument: /mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science/Titanic.ipynb
Info 08:47:41: TargetController found ID: .jvsc74a57bd097cc609b13305c559618ec78a438abc56230b9381f827f22d070313b9a1f3777./bin/python3.10./bin/python3.10.-m#ipykernel_launcher for document /mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science/Titanic.ipynb
Info 08:47:41: Setting controller affinity for /mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science/Titanic.ipynb .jvsc74a57bd097cc609b13305c559618ec78a438abc56230b9381f827f22d070313b9a1f3777./bin/python3.10./bin/python3.10.-m#ipykernel_launcher
Error 08:47:42: Failed to get activated enviornment variables ourselves Error: Command failed: . activate /home/minsa110/miniconda3/envs/golden-scenario && echo 'e8b39361-0157-4923-80e1-22d70d46dee6' && python /home/minsa110/.vscode-server/extensions/ms-toolsai.jupyter-2021.11.1001550889/pythonFiles/printEnvVariables.py
bash: activate: No such file or directory

    at ChildProcess.exithandler (child_process.js:308:12)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1048:16)
    at Socket.<anonymous> (internal/child_process.js:439:11)
    at Socket.emit (events.js:315:20)
    at Pipe.<anonymous> (net.js:673:12) {
  killed: false,
  code: 1,
  signal: null,
  cmd: ". activate /home/minsa110/miniconda3/envs/golden-scenario && echo 'e8b39361-0157-4923-80e1-22d70d46dee6' && python /home/minsa110/.vscode-server/extensions/ms-toolsai.jupyter-2021.11.1001550889/pythonFiles/printEnvVariables.py"
}
Info 08:47:43: Starting kernel from scratch
Error 08:47:43: Failed to get activated conda env variables for ~/miniconda3/envs/golden-scenario/bin/python
Info 08:47:43: Process Execution: > ~/miniconda3/envs/golden-scenario/bin/python -m pip list
> ~/miniconda3/envs/golden-scenario/bin/python -m pip list
Info 08:47:44: Registering dummy command feature
Warn 08:47:45: StdErr from Kernel Process /home/minsa110/anaconda3/envs/py38/lib/python3.8/site-packages/traitlets/traitlets.py:2202: FutureWarning: Supporting extra quotes around strings is deprecated in traitlets 5.0. You can use 'hmac-sha256' instead of '"hmac-sha256"' if you require traitlets >=5.

Warn 08:47:45: StdErr from Kernel Process   warn(

Warn 08:47:45: StdErr from Kernel Process /home/minsa110/anaconda3/envs/py38/lib/python3.8/site-packages/traitlets/traitlets.py:2157: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use '5a18cf90-9e8c-4576-801f-af3751d1d08d' instead of 'b"5a18cf90-9e8c-4576-801f-af3751d1d08d"'.

Warn 08:47:45: StdErr from Kernel Process   warn(

Info 08:47:46: Raw session started and connected
Started kernel py38 (Python 3.8.12)
Info 08:47:46: Finished connecting 4e87243b-396d-4d76-9f41-f0d6f66430ff
Info 08:47:46: UpdateWorkingDirectoryAndPath in Kernel
Info 08:47:46: changeDirectoryIfPossible
Info 08:47:46: Executing (status idle) silently Code = import os\nimport sys\n%cd "/mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science"
Info 08:47:46: Executing silently Code (completed) = import os\nimport sys\n%cd "/mnt/c/Users/somin/Documents/Projects/Golden Scenario/220106/data-science"
Info 08:47:46: Executing (status idle) silently Code = %config Completer.use_jedi = False
Info 08:47:46: Executing silently Code (completed) = %config Completer.use_jedi = False
Info 08:47:46: Executing (status idle) silently Code = %config InlineBackend.figure_formats = ['png']
Info 08:47:46: Executing silently Code (completed) = %config InlineBackend.figure_formats = ['png']
Info 08:47:46: Executing (status idle) silently Code = import sys\nprint(sys.executable)
Info 08:47:46: Executing silently Code (completed) = import sys\nprint(sys.executable)
Info 08:47:46: Waiting for idle on (kernel): ff3aea58-809b-46e3-a32e-5f9e5158629e -> idle
Info 08:47:46: Finished waiting for idle on (kernel): ff3aea58-809b-46e3-a32e-5f9e5158629e -> idle
Info 08:47:46: IPyWidgetScriptSource.initialize
Info 08:47:48: Registering dummy command feature
```

</p>
</details>

**Comment by greazer** [comment]
It's understood that an attempt to address this problem using a file watcher was problematic. But it still seems that there could be something Pylance could do to help the user "fix" the situation. Perhaps adding some codelens to cause a refresh to occur? We can chat about it.

**Comment by heejaechang** [comment]
it sounds like pylance server didn't see anything changed so it didnt refersh/restart the server and that is why it didn't get resolved.

currently pylance has a issue of detecting third party library installed/updated outside of workspace root for regular python file so it is probably same issue.

that being said, if restart/reloading windows is too much work, reopening notebook could send pylance server assosiciated witht he ntoebook to restart. that should fix the issue.

**Comment by heejaechang** [comment]
@minsa110 is this still a problem?

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.6.30, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2022630-22-June-2022)

**Comment by JhonAndersonVelasco** [comment]
I have the 2022.7.40 version with the same problem
